### PR TITLE
feat: #WB-1984, Add image editor and link to imagemenu

### PR DESCRIPTION
**Pull request by nabil-mansouri**
# Description

Ce commit intègre l'éditeur d'image et le branche à la barre de retouche d'image

>

## Type of change

Please check options that are relevant.

- [ ] Chore (PATCH)
- [ ] Doc (PATCH)
- [ ] Bug fix (PATCH)
- [X ] New feature (MINOR)
- [ ] Breaking change (MAJOR)

> PATCH: refactor, internal or non-breaking change which fixes an issue
>
> MINOR: non-breaking change which adds functionality
>
> MAJOR: fix or feature that would cause existing functionality to not work as expected

## Checklist:

- [X ] My code follows the style guidelines of this project
- [X ] I have performed a self-review of my own code
- [X ] I have commented my code, particularly in hard-to-understand areas
- [X ] My changes generate no new warnings

### CODE REVIEW

File: frontend/vite.config.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export default ({ mode }: { mode: string }) => {
     ? {
         "set-cookie": [
           `oneSessionId=${envs.VITE_ONE_SESSION_ID}`,
-          `XSRF-TOKEN=${envs.VITE_XSRF_TOKEN}`,
+          `XSRF-TOKEN=${envs.VITE_XSRF_TOKEN}; Path=/`,
```
Comment (ccreusat) [inline frontend/vite.config.ts:20]: ?

File: frontend/package.json [hunk 2]
```diff
@@ -47,45 +47,49 @@
...
+    "@tiptap/extension-table-row": "2.0.4",
+    "@tiptap/extension-text-align": "2.0.4",
+    "@tiptap/extension-text-style": "2.0.4",
+    "@tiptap/extension-typography": "2.0.4",
+    "@tiptap/extension-underline": "2.0.4",
+    "@tiptap/pm": "2.0.4",
+    "@tiptap/react": "2.0.4",
+    "@tiptap/starter-kit": "2.0.4",
+    "edifice-bootstrap": "develop-b2school",
+    "edifice-ts-client": "^1.0.0",
```
Comment (ccreusat) [inline frontend/package.json:85]: @nabil-mansouri peut-être regénérer le package.json avec le script `node scripts/package.cjs` car ta version de ts-client est pas bonne (1.0.0)

**Review by ccreusat — CHANGES_REQUESTED** [review]

File: frontend/src/components/Tiptap/Tiptap.tsx [hunk 3]
```diff
@@ -566,11 +608,24 @@ const Tiptap = () => {
       />
       {editor && (
         <BubbleMenu
-          shouldShow={({ editor }) => editor.isActive("custom-image")}
+          className={isImageModalOpen ? "d-none" : ""}
+          shouldShow={({ editor }) => {
+            console.log(editor.isActive("custom-image"), !isImageModalOpen);
```
Comment (Romu-C) [inline frontend/src/components/Tiptap/Tiptap.tsx:613]: console log?

File: frontend/vite.config.ts (hunk 1 above)
Comment (nabil-mansouri) [reply frontend/vite.config.ts:20]: je retesterais sans mais je sais pas pk j'avais des soucis de xsrf car il ajoutait le cookie sur tout les path sauf root...

**Pull request merged**